ox_profile/core/sampling.py

```diff
@@ -3,11 +3,44 @@
 
 import doctest
 import logging
+import platform
 import sys
 
 from ox_profile.core import metrics
 
 
+class Freezer(object):
```

Comment:
Great idea putting this logic into a context manager. Adding python 2.7 support is icing on the cake!